docassemble/assemblylinewizard/interview_generator.py

```diff
@@ -887,97 +891,140 @@ def get_docx_variables( text ):
-  """For a given set of specific cases, transform a
-  PDF field name into a standardized object name
-  that will be the value for the attachment field."""
-  if document_type.lower() == "docx":
-      return label # don't transform DOCX variables
-    
-  # Remove multiple appearance indicator, e.g. '__4' of 'users__4'
-  label = remove_multiple_appearance_indicator(label)
+def map_names(label, document_type="pdf", reserved_whole_words=generator_constants.RESERVED_WHOLE_WORDS,
+              reserved_prefixes=generator_constants.RESERVED_PREFIXES,
+              reserved_var_plurals=generator_constants.RESERVED_VAR_PLURALS,
+              reserved_pluralizers_map = generator_constants.RESERVED_PLURALIZERS_MAP,
+              reserved_suffixes_map=generator_constants.RESERVED_SUFFIXES_MAP):
+    """For a given set of specific cases, transform a
+    PDF field name into a standardized object name
+    that will be the value for the attachment field."""
+    if document_type.lower() == "docx":
+        return label # don't transform DOCX variables
 
-  if exactly_matches_reserved_word(reserved_whole_words, label):
-    return label
+    # Remove multiple appearance indicator, e.g. '__4' of 'users__4'
+    label = remove_multiple_appearance_indicator(label)
 
-  # For the sake of time, this is the fastest way to get around something being plural (wrap it in `str()`)
-  if is_a_plural(reserved_var_plurals, label):
-    return get_stringifiable_version(label)
+    if label in reserved_whole_words:
+        return label
+
+    if label in reserved_var_plurals:
+        return label
+
+    # Break up label into its parts
+    label_groups = get_reserved_label_parts(reserved_prefixes, label)
+
+    # If no matches to automateable labels were found,
+    # just use the label as it is
+    if label_groups is None or label_groups[1] == '':
+        return label
+
+    # With reserved words, we're always using an index
+    # of the plural version of the prefix of the label
+    prefix = label_groups[1]
+    # Turn any singluars into plurals, e.g. 'user' into 'users'
+    plural_prefix = reserved_pluralizers_map[prefix]
+    digit = label_groups[2]
+    index = indexify(digit)
+    # it's just a standalone, like "defendant", or it's a numbered singular
+    # prefix, e.g. user3
+    if label == prefix or label == prefix + digit: 
+        return plural_prefix + index # the vars w/ the index, users[0] or users[2]
+
+    suffix = label_groups[3]
+    # Avoid transforming arbitrary suffixes into attributes
+    if not suffix in reserved_suffixes_map:
+        return label  # return it as is
+
+    # Get the mapped suffix attribute if present, else just use the same suffix
+    suffix_as_attribute = reserved_suffixes_map.get(suffix, suffix)
+    return "".join([plural_prefix, index, suffix_as_attribute])
+
+
+def trigger_gather_string(docassemble_var, 
+                          docx_only_suffixes=generator_constants.DOCX_ONLY_SUFFIXES,
```

Comment:
For future issue, some `docx_only_suffixes` constants are missing leading `.`s: https://github.com/SuffolkLITLab/docassemble-assemblylinewizard/blob/master/docassemble/assemblylinewizard/generator_constants.py#L153-L154

Reply:
Does not affect current code since the items on the list can still match inside the variable's suffix.

Reply:
Good call, I'll make a separate PR really quick (would take just as long to make as an issue).

Reply:
Can we make sure this is correct? Is this used as a regex or as an exact string? `.` means match any character, which is NOT the same as match only a `.`. Might need to escape it like this: `\.`

Reply:
Good call, I didn't recognize those were regexes. Probably should have because of the raw strings. Should be fixed in https://github.com/SuffolkLITLab/docassemble-assemblylinewizard/pull/144/commits/9c088f7fec9762c93ab30278c56a05aeadd59e33

Reply:
I'd understood that that global variable contained strings, not regex strings

Reply:
You are right, but I think it's only used in this function (and in `is_reserved_docx_label`), but in there it's directly concatenated into  `docx_only_suffixes_regex`, without processing those strings to make them regex ready.